File: USACO/USOpen/s1.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    FILE *fin, *fout;
    fin = fopen("sort.in","r");
    fout = fopen("sort.out","w+");
    int N; fscanf(fin,"%d",&N);
    int A[N];
    for(int i = 0; i < N; ++i)
        fscanf(fin,"%d",A+i);
    int cnt = 0;
    bool sorted = false;
    while(!sorted){
        sorted=true;
        cnt++;
        for(int i = 0; i < N-1; ++i){
            if(A[i+1] < A[i]){
                int t = A[i+1];
                A[i+1] = A[i];
                A[i] = t;
                sorted=false;
            }
        }
    }
    fprintf(fout,"%d\n",cnt);
    fclose(fin);
    fclose(fout);
    return 0;
}

File: USACO/USOpen/s1.py
fin = open("sort.in","r")
fout = open("sort.out","w+")
N = int(fin.readline())
A = [int(fin.readline()) for _ in xrange(N)]
sA = sorted(A)
pos = {}
cnt = {}
for i,x in enumerate(sA):
    if x not in pos:
        pos[x] = i

mx = 0
for i,x in enumerate(A):
    finalPos = pos[x] + (cnt[x] if x in cnt else 0)
    if x in cnt:
        cnt[x]+=1
    else:
        cnt[x] = 1
    dist = i - finalPos
    mx = max(mx,dist)

fout.write(str(mx+1)+"\n")

fin.close()
fout.close()
